src/server/object_services/object_server.js

```diff
@@ -983,6 +983,39 @@ async function delete_multiple_objects_by_filter(req) {
     return { is_empty: !bucket_has_objects };
 }
 
+/**
+ * delete_multiple_objects_unordered is an internal function which
+ * takes a number `limit` and a `bucket_id` and will delete the `limit`
+ * objects from the bucket in NO PARTICULAR ORDER.
+ * 
+ * This function is inteded to use in the case of a bucket deletion
+ * where we want to delete all the objects in the bucket but we don't
+ * care about the order in which they are deleted or the versioning, etc.
+ * @param {*} req 
+ */
+async function delete_multiple_objects_unordered(req) {
+    load_bucket(req, { include_deleting: true });
+    const bucket_id = req.bucket._id;
+    const limit = req.rpc_params.limit;
+    dbg.log0(`delete_multiple_objects_unordered: bucket=${req.bucket.name} limit=${limit}`);
+
+    // find the objects - no need to paginate explicitly because
+    // find_objects will ensure that it does not return any object
+    // which is already marked for deletion and that's all we care
+    // about here.
+    const { objects } = await MDStore.instance().find_objects({
+        bucket_id: make_md_id(bucket_id),
+        limit,
+        key: undefined,
+    });
+
+    // delete the objects
+    await MDStore.instance().remove_objects_and_unset_latest(objects);
+
+    const bucket_has_objects = await MDStore.instance().has_any_objects_for_bucket(bucket_id);
```

Comment:
We can remove this and just check the objects.length.
Both here and also in the previous function delete_multiple_objects_by_filter().